src/options.rs

```diff
@@ -145,25 +204,26 @@ impl Options {
 					return Err(Error::InvalidConfiguration(format!(
 						"Unsupported database version {}. Expected {}", version, CURRENT_VERSION)));
 				}
+				if version == 3 {
+					//Treat all tables as ref counted.
+					force_rc = true;
```

Comment:
I am not sure the previous change in rc will migrate successfully.
There can be entry with size above the new default max default table size for whom we will read an incorrect size.

Reply:
These should produce a decompression error, which I this is fine for now. I guess we could additionally check that compression bit is not set for columns with no `compression` option as well

Reply:
Only if there is a compression (the bit is also used for NoCompression to allow switching). Additional check seems worth it.

Reply:
Will add in the next PR